src/common/snippets/include/snippets/runtime_configurator.hpp

```diff
@@ -109,12 +120,70 @@ class RuntimeConfigurator {
     void update_buffer_scratchpad_size(const lowered::LinearIRPtr& linear_ir) const;
     /**
      * @brief Calculate data offsets of LinearIR and update these values in RuntimeConfig
+     * @param shapes shapes used in offsets computation
+     * @param layouts layouts used in offsets computation
+     */
+    void update_data_offsets(const std::vector<ov::snippets::VectorDims>& shapes,
+                             const std::vector<std::vector<size_t>>& layouts) const;
+    /**
+     * @brief Extract shapes from m_io_descs
      */
-    void update_data_offsets() const;
+    std::vector<ov::snippets::VectorDims> extract_shapes();
     /**
-     * @brief Update latest input shapes
+     * @brief Extract layouts from m_io_descs
      */
-    void update_latest_shapes();
+    std::vector<std::vector<size_t>> extract_layouts();
+
+    class ParallelWAOptimizer {
+    public:
+        /**
+         * @brief Inits ParallelWAOptimizer: computes optimizer parameters which should be set at compilation stage
+         * @param linear_ir LinearIR
+         */
+        void init(const ov::snippets::lowered::LinearIRPtr& linear_ir,
+                  const std::vector<snippets::lowered::PortDescriptorPtr>& io_descs,
+                  size_t in_num);
+        /**
+         * @brief Defines if the optimizer should be applied
+         * @attention It also computes "batch_m" and "new_m" runtime parameters
+         * @param master_shape Master shape
+         */
+        bool need_optimize(const ov::snippets::VectorDims& master_shape);
+        /**
+         * @brief Updates all the necessary runtime information
+         * @param map Loop info -> Runtime params map which will be passed in "update_loop_info"
+         * the map is filled with updated loops_to_split loops: "new_m" work amount is set for them, and runtime params are updated correspondingly
+         * @param shapes Vector which is filled with the split shapes
+         * @param layouts Vector which is filled with the split layouts
+         * @param config Config which should be updated: tile rank and master shape is updated
+         * @attention It also computes "batch_m" and "new_m" runtime parameters
+         * @param master_shape Master shape
+         */
+        void update(ov::snippets::RuntimeConfigurator::LoopInfoRuntimeParamsMap& map,
+                    std::vector<ov::snippets::VectorDims>& shapes,
+                    std::vector<std::vector<size_t>>& layouts,
+                    const std::shared_ptr<ov::snippets::RuntimeConfig>& config,
+                    size_t in_num);
```

Comment:
As an alternative option, we could have only one `optimize` method with 6 params: master shape (which is used to identify if the optimization should be applied), and the rest 5 params that we already have for `update`.
Thanks to that, we would probably avoid `batch_m` and `new_m` storage as class members
@a-sidorova what do you think?

Reply:
I think it makes sense in general 🤔 
Good idea, thank you!

Reply:
Done, please check